lib/galaxy/config_watchers.py

```diff
@@ -81,11 +79,11 @@ def start(self):
         if self.app.config.config_file:
             self.core_config_watcher.watch_file(
                 self.app.config.config_file,
-                callback=partial(reload_config_options, self.app.config)
+                callback=lambda path: self.app.queue_worker.send_control_task('reload_core_config')
```

Comment:
This will raise an error: watcher invokes the callback with an argument: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/util/watcher.py#L175

Reply:
Why, you're not using `path` ?

Reply:
hmm. My comment just disappeared (was editing it after posting). Here it is, just in case:
This will raise an error: watcher invokes the callback with an argument: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/util/watcher.py#L175
(i agree, doesn't look good, but we'd need to refactor watcher most likely).
And no, we don't need path for reloading config options.

Reply:
so either I don't know my python, but `lambda path: self.app.queue_worker.send_control_task('reload_core_config')`

is the same as
```
def lambda(path):
    return self.app.queue_worker.send_control_task('reload_core_config')
```
So I don't understand why you want to include `path` in the lambda ?

Reply:
The line you quote doesn't look relevant, the path variable there comes from a watchdog event: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/util/watcher.py#L149

Reply:
Well, you *do* know your python, so I suspect it's just me not looking deep enough. But here's what I get when I test it (enable, then edit galaxy.yml):

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/sergey/0dev/galaxy/galaxy/current/.venv/local/lib/python2.7/site-packages/watchdog/observers/api.py", line 199, in run
    self.dispatch_events(self.event_queue, self.timeout)
  File "/home/sergey/0dev/galaxy/galaxy/current/.venv/local/lib/python2.7/site-packages/watchdog/observers/api.py", line 368, in dispatch_events
    handler.dispatch(event)
  File "/home/sergey/0dev/galaxy/galaxy/current/.venv/local/lib/python2.7/site-packages/watchdog/events.py", line 322, in dispatch
    self.on_any_event(event)
  File "lib/galaxy/util/watcher.py", line 137, in on_any_event
    self._handle(event)
  File "lib/galaxy/util/watcher.py", line 175, in _handle
    callback(path=path)
TypeError: <lambda>() takes no arguments (1 given)
```
I'm heading to the office, will look into  it once I get there (back online in ~50min)

Reply:
Ah, I see. `path` isn't used, but the the event handler passes it. My bad.